src/Application/UI/Control.php

```diff
@@ -42,6 +45,23 @@
 	}
 
 
+	final public function setTemplateFile(string $templateFile = null)
+	{
+		$this->templateFile = $templateFile;
+
+		if ($this->template !== null) {
```

Comment:
Conditional setter? Never seen that before. Should throw exception or create template if not already created.
I understand why it's here, but it's a just a very specific usecase.

Reply:
I don't know if you got it right. Whole idea of this PR is to be able setup file without creating template. If we throw exception or create template there is no PR needed any more, cause it will work just like shortcut for `$control->getTemplate()->setFile()`. 

But I thought about it if I should add this condition. Because it's just for specific usecase as you said when user try to set file after template was created to prevent "WTF?" situations and let setter work like user expect.

Reply:
As I said, I understand. But imagine you don't know that code, just api. It looks exactly like a shortcut, but acts absolutely diferent.

Reply:
I don't, I'm not saying you're not right, but I think user don't care about it, he just want to set template file/view and this did that without unnecessary loads.

And I really hate setting template in `$onAnchor` event. Or do you know way how to do it without event in control factory or in `createComponent<name>()` method?

Reply:
I think you could configure template file through it's factory definition
```
fooControlFactory:
  implement IFooControlFactory
  setup:
    - getTemplate()->setFile('/path/to/file')
```

But I never used that approach. I have ITemplate with setView method and to change path to these views I only need change pattern in configuration.

Reply:
This will end with `InvalidStateException` or new template instance. Anyway I don't like to setup file template in config, there are situations where you want to setup view by condition, or you have one control with factory and want to setup many different view for actual page.